sdk/identity/azure-identity-v2/src/main/java/com/azure/identity/v2/implementation/client/ClientBase.java

```diff
@@ -55,13 +52,36 @@ HttpPipeline getPipeline() {
     }
 
     HttpPipeline setupPipeline() {
-        //TODO (g2vinay): Wire the HttpPipelineOptions in Pipeline construction.
+        // Closest to API goes first, closest to wire goes last.
         List<HttpPipelinePolicy> policies = new ArrayList<>();
-        policies.add(new HttpRetryPolicy());
-        HttpPipeline httpPipeline = new HttpPipelineBuilder().addPolicy(policies.get(0))
-            .httpClient(getHttpPipelineOptions().getHttpClient())
-            .build();
-        return httpPipeline;
+
+        policies.add(clientOptions.getHttpPipelineOptions().getHttpRedirectOptions() == null
```

Comment:
just for readability can we pull `clientOptions.getHttpPipelineOptions()` into a local?